src/common/meson.build

```diff
@@ -0,0 +1,384 @@
+#else
+	int name[2] = { CTL_HW, HW_NCPU };
+#endif
+	int cpus;
+	size_t cpus_size = sizeof(cpus);
+	sysctl(name, 2, &cpus, &cpus_size, NULL, 0);
+	return 0;
+}
+'''
+
+cpu_sysconf_src = '''
+#include <unistd.h>
+int
+main(void)
+{
+	long i;
+#ifdef _SC_NPROCESSORS_ONLN
+	/* Many systems using sysconf() */
+	i = sysconf(_SC_NPROCESSORS_ONLN);
+#else
+	/* IRIX */
+	i = sysconf(_SC_NPROC_ONLN);
+#endif
+	return 0;
+}
+'''
+
+cpu_pstat_getdynamic_src = '''
+#include <sys/param.h>
+#include <sys/pstat.h>
+
+int
+main(void)
+{
+	struct pst_dynamic pst;
+	pstat_getdynamic(&pst, sizeof(pst), 1, 0);
+	(void)pst.psd_proc_cnt;
+	return 0;
+}
+'''
+
+if cc.compiles(cpu_special_src, name : 'number of cpu cores on Windows or Cygwin')
+# nothing to do, here
+elif cc.compiles(cpu_sched_getaffinity_src, name : 'number of cpu cores with sched_getaffinity()')
+    config_h.set10('TUKLIB_CPUCORES_SCHED_GETAFFINITY', true)
+elif cc.compiles(cpu_cpuset_src, name : 'number of cpu cores with cpuset()')
+    config_h.set10('TUKLIB_CPUCORES_CPUSET', true)
+elif cc.compiles(cpu_sysctl_src, args : have_sys_param_h ? '-DHAVE_SYS_PARAM_H' : '', name : 'number of cpu cores with sysctl()')
+    config_h.set10('TUKLIB_CPUCORES_SYSCTL', true)
+elif cc.compiles(cpu_sysconf_src, name : 'number of cpu cores with sysconf()')
+    config_h.set10('TUKLIB_CPUCORES_SYSCONF', true)
+elif cc.compiles(cpu_pstat_getdynamic_src, name : 'number of cpu cores with pstat_getdynamic()')
+    config_h.set10('TUKLIB_CPUCORES_PSTAT_GETDYNAMIC', true)
+endif
+
+# tuklib_integer.m4
+
+bswap_src = '''
+int main()
+{
+	__builtin_bswap16(1);
+	__builtin_bswap32(1);
+	__builtin_bswap64(1);
+	return 0;
+}
+'''
+res = cc.links(bswap_src, name: '__builtin_bswap16/32/64')
+if res
+    config_h.set10('HAVE___BUILTIN_BSWAPXX', true,
+        description: 'Define to 1 if the GNU C extensions __builtin_bswap16/32/64 are supported.'
+        )
+else
+    have_byteswap_h = false
+    have_sys_endian_h = false
+    have_sys_byteorder_h = false
+    bswap_headers = [
+        'byteswap.h',
+        'sys/endian.h',
+        'sys/byteorder.h'
+    ]
+    foreach h : bswap_headers
+        if cc.has_header(h)
+            config_h.set10('HAVE_' + h.underscorify().to_upper(), true)
+            set_variable(f'have_' + h.underscorify(), true)
+        endif
+    endforeach
+
+    if have_byteswap_h
+        bswap_fcts = [
+            'bswap_16',
+            'bswap_32',
+            'bswap_64'
+        ]
+        foreach f : bswap_fcts
+            bswap_src = '''
+#include <byteswap.h>
+int main()
+{
+	@0@(42);
+	return 0;
+}
+'''
+            if cc.links(bswap_src.format(f), name: f)
+                config_h.set10('HAVE_' + f.to_upper(), true,
+                    description: 'Define to 1 if ' + f + '() is available.'
+                )
+            endif
+        endforeach
+    endif
+endif
+
+fast_unaligned_access = false
+if get_option('unaligned-access').auto()
+    unaligned_access_cpus = [
+        'ppc',
```

Comment:
The Autoconf and CMake builds exclude 32-bit PowerPC from fast unaligned access as I'm not sure how it behaves, especially with 64-bit values. I suppose it's an uncommon target anyway so better play it safe.